src/sentry/release_health/metrics_sessions_v2.py

```diff
@@ -353,31 +434,103 @@ def run_sessions_query(
     return json_to_snql(
         {"selected_columns": ["value"], "conditions": conditions}, entity=dummy_entity
     ).where
 
 
-def _parse_orderby(query: QueryDefinition) -> Optional[OrderBy]:
+def _transform_conditions(
+    conditions: ConditionGroup,
+) -> Tuple[ConditionGroup, StatusFilter]:
+    """Split conditions into metrics conditions and a filter on session.status"""
+    if not conditions:
+        return conditions, None
+    where, status_filters = zip(*map(_transform_single_condition, conditions))
+    where = [condition for condition in where if condition is not None]
+    status_filters = [f for f in status_filters if f is not None]
+    if status_filters:
+        status_filters = frozenset.intersection(*status_filters)
+    else:
+        status_filters = None
+    return where, status_filters
+
+
+def _transform_single_condition(
+    condition: Union[Condition, BooleanCondition]
+) -> Tuple[Optional[Union[Condition, BooleanCondition]], StatusFilter]:
+    if isinstance(condition, Condition):
+        if condition.lhs == Function("ifNull", parameters=[Column("session.status"), ""]):
+            # HACK: metrics tags are never null. We should really
+            # write our own parser for this.
+            condition = replace(condition, lhs=Column("session.status"))
+        if condition.lhs == Column("session.status"):
+            if condition.op == Op.EQ:
+                return None, _parse_session_status(condition.rhs)
+            if condition.op == Op.NEQ:
+                return None, ALL_STATUSES - _parse_session_status(condition.rhs)
+            if condition.op == Op.IN:
+                return None, frozenset.union(
+                    *[_parse_session_status(status) for status in condition.rhs]
+                )
+            if condition.op == Op.NOT_IN:
+                return None, ALL_STATUSES - frozenset.union(
+                    *[_parse_session_status(status) for status in condition.rhs]
+                )
+            raise InvalidParams("Unable to resolve session.status filter")
+
+    if "session.status" in str(condition):
+        # Anything not handled by the code above cannot be parsed for now,
+        # for two reasons:
+        # 1) Queries like session.status:healthy OR release:foo are hard to
+        #    translate, because they would require different conditions on the separate
+        #    metric fields.
+        # 2) AND and OR conditions come in the form `Condition(Function("or", [...]), Op.EQ, 1)`
+        #    where [...] can again contain any condition encoded as a Function. For this, we would
+        #    have to replicate the translation code above.
+        raise InvalidParams("Unable to parse condition with session.status")
+
+    return condition, None
+
+
+def _parse_session_status(status: Any) -> FrozenSet[SessionStatus]:
+    try:
+        return frozenset([SessionStatus(status)])
+    except ValueError:
+        return frozenset()
```

Comment:
Should we add some logging/tracking/whatever here? It may ease potential debugging in the future.

Reply:
This is actually normal behavior consistent with other filters: Filtering by `release:foo` will give you empty results, and so will `session.status:foo`. So we don't need to track it.